docs/chapter4-payload-handoff-format.rst

```diff
@@ -29,9 +29,478 @@ architecture, either 32 or 64 bits.
+.. table:: /pci/pci-rb@<ecam-base-address>
+
+   ================= =========== ==================== ======================================================================================================================================================================================================================================================================================================================================================================================================================================
+   Property Name     Usage       Value Type           Definition
+   ================= =========== ==================== ======================================================================================================================================================================================================================================================================================================================================================================================================================================
+   compatible        R           string list          To determine what kind of driver should be applied to the device. It should be in the form of <manufacturer>,<model>. But for historical reason,
+                                                      <model> alone also works for common used devices, in this spec, "pci-host-ecam-generic" is used for generic support.
+   #address-cells    R           u32                  Indicates the number of cells used in addresses in the handoff.
+
+                                                      Per PCI Bus Binding spec [PCIBusBinding]_, **this value must be 3**.
+   #size-cells       R           u32                  Indicates the number of cells used for sizes in the handoff.
+
+                                                      Per PCI Bus Binding spec [PCIBusBinding]_, **this value must be 2**.
+   bus-range         R           prop-encoded-array   Contains 2 cells, each encoded as with encode-int, the first representing the bus number of the PCI bus implemented by the bus controller
+                                                      represented by this node (the secondary bus number in PCI-to-PCI bridge nomenclature), and the second representing the largest bus number of any
+                                                      PCI bus in the portion of the PCI domain that is subordinate to this node (the subordinate bus number in PCI-to-PCI bridge nomenclature).
+
+                                                      This is required for Payload to do a simple PCI scan to build a PCI device database for providing functionality/service drivers (graphics, USB,
+                                                      SATA…)
+   reg               R           prop-encoded-array   indicates an address range that are used by the PCI devices under the current bridge. It is a range that the CPU (parent bus) can access. The
```

Comment:
Please update description that this is PCIEXBAR which is ECAM configuration space.

Reply:
Looks like this is not PCIEXBAR, it looks more like MMIO or IO regions belong to current PCIe device. 
see example here: https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/pci/ti-pci.txt#L78

However, with address-cells = <1>, how do we know the base address type? MMIO or IO?

Reply:
So I think the reg property is always specified in the address space of the parent, not the child so whatever cell properties have been defined at the parent level there are applied. In this case ECAM is just a memory mapped region so I'd suspect 2 addr and 2 size cells for 64bit MEM address space. The address-cells and size-cells on this node apply to the ranges which follow https://www.openfirmware.info/data/docs/bus.pci.pdf which seems to be what is done below.

Reply:
Yes that looks right to me

Reply:
there are some mistakes on the reg, i have updated, please take a look.

Reply:
LGTM